indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceIngestionSpec.java

```diff
@@ -48,7 +51,15 @@ public DatasourceIngestionSpec(
   )
   {
     this.dataSource = Preconditions.checkNotNull(dataSource, "null dataSource");
-    this.interval = Preconditions.checkNotNull(interval, "null interval");
+
+    List<Interval> theIntervals = null;
+    if (interval != null) {
+      theIntervals = ImmutableList.of(interval);
+    } else if (intervals != null && intervals.size() > 0) {
+      theIntervals = JodaUtils.condenseIntervals(intervals);
+    }
```

Comment:
I think it's always good to provide a transition period where both are supported, as this makes it a lot easier to migrate.

1. update druid to a version that supports both old and new forms 
2. switch your client to produce the new form
3. later on, you can update druid to a version that only supports the new one